shop-catalog/frontends/react-redux/src/components/pagination/pagination.tsx

```diff
@@ -1,76 +1,79 @@
 import React from "react";
 
 import ZenCore from "@zenflux/core";
-import ZenRedux from "@zenflux/redux";
 
 import { ExtendType, IPaginationProps } from "./model";
 
 import "./pagination.css";
 
 export default function Pagination( props: IPaginationProps ): JSX.Element {
-    const offsetState = React.useState( 0 ),
+    const isRendered = React.useRef( false );
+
+    const offsetState = React.useState( -1 ),
         [ offset, setOffset ] = offsetState;
 
-    const prevOffset = ZenRedux.hooks.usePrevious( offset );
+    let newOffset = offset;
 
     const onExtend = ( type: ExtendType ) => {
-        ZenCore.managers.commands.run( "Components/Pagination/Controller/Extend", {
+        ZenCore.managers.commands.run( "Components/Pagination/Controller/Commands/Extend", {
             type,
-            offsetState,
-            maxVisiblePages: props.maxVisiblePages,
+            setOffset,
+            offset: newOffset,
+            range: props.maxVisiblePages,
             controller: props.controller,
         } );
     };
 
     const onChange = ( page: number ) => {
-        ZenCore.managers.commands.run( "Components/Pagination/Controller/Set", {
+        ZenCore.managers.commands.run( "Components/Pagination/Controller/Commands/Set", {
             page,
             controller: props.controller,
         } );
     };
 
-    const inRange = function ( active: number, min: number, max: number ) {
-        return ( active >= min && active < max );
-    };
+    // Used to update the offset from `extend` command, when new offset is 0.
```

Comment:
```suggestion
    // Since each page change, pagination may re-render, to back to lower offset of 0 will be not possible if the default offset is 0.
```